tracing-subscriber/src/fmt/format/json.rs

```diff
@@ -236,6 +236,27 @@ where
                 serializer.serialize_entry("target", meta.target())?;
             }
 
+            if self.display_thread_name {
+                match std::thread::current().name() {
+                    Some(name) => {
+                        serializer.serialize_entry("threadName", name)?;
+                    }
+                    // fall-back to thread id when name is absent and ids are not enabled
+                    None if !self.display_thread_id => {
+                        serializer.serialize_entry(
+                            "threadName",
+                            &format!("{:?}", std::thread::current().id()),
+                        )?;
+                    }
+                    _ => {}
```

Comment:
Nit/TIOLI: Might be worth storing `thread::current()` in a `let` binding here so that we don't get it twice in the casxe where the thread has no name.